draft-trammell-rains-protocol.md

```diff
@@ -911,56 +890,49 @@ The value of the context (6) key is a UTF-8 encoded string containing the name
-entire lexicographic space of subject names within the Zone, or it may be
-organized as Singular Assertions. 
-
-Within the shards array, if present, the contained Shards MUST be sorted in
-lexicographic order by shard range start. Within the assertions array, if
-present, the contained Singular Assertions MUST be sorted in lexicographic order
-by subject name.
+The value of the assertions (20) key is a CBOR array of Singular Assertions as
+defined in {{singular-assertions}}. The CBOR array contains all Singular
+Assertions of this zone and context and they MUST be sorted (see {{c14n}}).
 
 ### P-Shards {#p-shards}
 
+[EDITOR's NOTE: use consistently nonexistence or non-existence]
+
 Shards ({{shards}}) can be used as definitive proof of the nonexistence of a
 name within a zone. P-Shards serve the same purpose, but offer only a
-probabalistic guarantee of the non-existence of the name. Specifically, as they
+probabilistic guarantee of the non-existence of a name. Specifically, as they
 are based on Bloom filters, a subject name which does not in fact exist may
 appear in the P-Shard; in return for this uncertainty, they offer a much more
-space-efficient way to demonstrate the non-existence of a subject within the
-zone than Shards do. There is a tradeoff between the size of the bit string
-storing the Bloom filter, the number of Assertions covered by the P-Shard, and
-the false positive error rate. The zone authority can determine how to weight
-them.
+space-efficient way to demonstrate the non-existence of an Object for a subject
+name within the zone and context than Shards do. There is a tradeoff between the
+size of the bit string storing the Bloom filter, the number of Assertions
+covered by the P-Shard, and the false positive error rate. The zone Authority
+can determine how to weight them.
 
 A P-Shard is represented as a CBOR map. This map MUST contain the signatures
 (0), subject-zone (4), context (6), and content(23) keys. It MAY contain the
 range(11) key.
 
 The value of the signatures (0) key is an array of one or more Signatures as
 defined in {{signatures}}. The signatures on the P-Shard are to be verified
-against the appropriate key for the Zone for which the P-Shard is cvalid in the
+against the appropriate key for the Zone for which the P-Shard is valid in the
 given context.
 
 The value of the subject-zone (4) key is a UTF-8 encoded string containing the
-name of the zone in which the Assertions in the P-Shard is made and MUST end
-with '.' (the root zone).
+name of the zone in which the Singular Assertions in the P-Shard are made and
+MUST end with '.' (the root zone).
 
 The value of the context (6) key is a UTF-8 encoded string containing the name
-of the context in which the Assertions in the P-Shard are valid. Both the
-authority-part and the context-part MUST end with a '.'.
+of the context in which the Singular Assertions in the P-Shard are valid. Both
```

Comment:
The P-Shard doesn't contain Singular Assertions, it contains a Bloom filter against which names and objects can be checked. Rework this.

Reply:
True, my bad...